README.md

```diff
@@ -1,7 +1,154 @@
+generated from the L1 transaction), which moves 10WABC from the L2 bridge to Alice's address on Obscuro. 
+
+This logic is part of the consensus of Obscuro, every node receiving the same block containing the rollup and the deposits, will generate the exact same synthetic transaction.
+
+See [go/enclave/bridge](go/enclave/bridge)
+
+*Note that the current bridge implementation is very primitive and only features two supported hardcoded ERC20 tokens to demonstrate
+the mechanics.*
+
+
+#### 5. Mempool 
+
+The mempool is the component which handles the incoming transactions and is responsible for selecting which transactions 
+to include in the current batch and pick the order.
+
+The big advantage of running the mempool inside the secure enclave is that the ordering of transactions cannot be gamed by the aggregator, 
+which makes MEV much more difficult.
+
+See [go/enclave/mempool](go/enclave/mempool)
+
+*Note that the current mempool implementation is very primitive. It always includes all received transactions that were not already
+included in a rollup.*
+
+
+#### 6. The rollups and the PoBI protocol
+
+Like in any blockchain the unit of the protocol is the batch of transactions organized in a chain. 
+The Obscuro blocks have an encrypted payload, which is only visible inside the secure enclave.
+All the logic of maintaining the current state based on incoming data and of producing new rollups is found in the
+[go/enclave/rollupchain](go/enclave/rollupchain) package.
+
+
+#### 7. Cryptography
+
+This is where the Obsuro specific cryptography is implemented. 
+
+- Master seed generation
+- Key derivation
+- Payload encryption/decryption
+
+
+See [go/enclave/crypto](go/enclave/crypto)
+
+*Note: The current implementation is still very incipient. There are hardcoded keys, no key derivation, etc.*
+
+
+### II. The Host
+
+The host service is the software that is under the control of the operator. It does not run inside a secure enclave, and there is no attestation on it.
+
+From a threat model point of view, the host service is seen as an adversary by an enclave. Any data that it feeds into the enclave
+will be verified and considered malicious.
+
+A secure solution that uses confidential computing will generally try to minimize the TCB, and run as much as possible outside the secure enclave,
+while still achieving the same security goals.
+
+The host service is the equivalent of a typical blockchain node, and is responsible for:
+ 
+- P2P messaging: Gossiping of encrypted transactions and rollups
+- RPC: Exposing an RPC interface similar to the one exposed by normal ethereum nodes
+- Communicating with an Ethereum node for retrieving blocks and for submitting transactions with data that was generated inside the enclave. 
+This means it has to be an ethereum wallet and control keys to accounts with enough Eth to publish transactions.  
```

Comment:
Ethereum and ETH.
This sentence does not read well. Perhaps "This means an Ethereum wallet and the control keys to accounts with enough ETH to publish transactions is required."